lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -112,6 +119,9 @@ impl ShardReplicaSet {
 
             LocalShard::move_data(replica_path, &self.shard_path).await?;
 
+            // remove shard_id initialization flag because shard is fully recovered
```

Comment:
We don't know if it is fully recovered before we actually load the shard. It might be recovered from corrupted snapshot